src/Swashbuckle.AspNetCore.SwaggerGen/XmlComments/XmlCommentsRecursion.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Globalization;
+using System.Xml.XPath;
+
+namespace Swashbuckle.AspNetCore.SwaggerGen;
+
+/// <summary>
+///     Service to handle recursive retrieval of XML comments from an XPathNavigator.
+/// </summary>
+internal static class XmlCommentsRecursionService
```

Comment:
This doesn't feel "service"-y to me - it's more "helper"-y. I don't think we need recursive in the name either - that feels like an implementation detail to me.

Also the file name should match the type name.